### PR TITLE
[Merged by Bors] - feat(Analysis/Distribution/SchwartzSpace): Constructor for CLMs to normed space and integration

### PR DESCRIPTION
**Pull request by mcdoll**
- We define a constructor for continuous linear maps from Schwartz space into normed spaces similar to mkCLM
- Define the integral as a distribution on Schwartz space (moved to lemmas from `Analysis/Distribution/FourierSchwartz` to `Analysis/Distribution/SchwartzSpace` for that)
- Golf a few definitions

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean [hunk 1]
```diff
@@ -1003,17 +1017,85 @@ theorem iteratedPDeriv_eq_iteratedFDeriv {n : ℕ} {m : Fin n → E} {f : 𝓢(E
...
+    _ = (1 + ‖x‖) ^ (finrank ℝ D + 1 + k) * ‖f x‖ := by simp only [pow_add, mul_assoc]
+    _ ≤ C := hC x
+
+lemma integrable (f : 𝓢(D, V)) {μ : Measure D} [IsAddHaarMeasure μ] :
+    Integrable f μ :=
+  (f.integrable_pow_mul (μ := μ) 0).mono f.continuous.aestronglyMeasurable
+    (eventually_of_forall (fun _ ↦ by simp))
+
+/-- The integral as a continuous linear map from Schwartz space to the codomain. -/
+def integral (μ : Measure D) [IsAddHaarMeasure μ] : 𝓢(D, V) →L[ℝ] V :=
```
Comment (loefflerd) [inline Mathlib/Analysis/Distribution/SchwartzSpace.lean:1056]: Maybe this should be called `integralCLM`, for consistency with other similar declarations?

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean [hunk 2]
```diff
@@ -1003,17 +1017,85 @@ theorem iteratedPDeriv_eq_iteratedFDeriv {n : ℕ} {m : Fin n → E} {f : 𝓢(E
...
+      let m := (l, 0)
+      use Finset.Iic (l, 0)
+      use 2^l * ∫ x : D, (1 + ‖x‖)^(- (l : ℝ)) ∂μ
+      have hpos : 0 ≤ ∫ x : D, (1 + ‖x‖)^(- (l : ℝ)) ∂μ := by
+        apply integral_nonneg
+        intro
+        positivity
+      refine ⟨by positivity, fun f ↦ (norm_integral_le_integral_norm f).trans ?_⟩
+      have h : ∀ x, ‖f x‖ ≤ (1 + ‖x‖)^(-(l : ℝ)) *
+          (2^l * ((Finset.Iic m).sup (fun m => SchwartzMap.seminorm ℝ m.1 m.2) f)) := by
```
Comment (loefflerd) [inline Mathlib/Analysis/Distribution/SchwartzSpace.lean:1071]: Using `m` in `fun m => ...` when you also have a local `let` variable `m` in the same line is confusing. There is also some nonstandard spacing here (`2^l` etc).

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean [hunk 3]
```diff
@@ -1003,17 +1017,85 @@ theorem iteratedPDeriv_eq_iteratedFDeriv {n : ℕ} {m : Fin n → E} {f : 𝓢(E
...
+        intro x
+        rw [rpow_neg (by positivity), ← div_eq_inv_mul, le_div_iff' (by positivity), rpow_nat_cast]
+        simpa using one_add_le_sup_seminorm_apply (m := m) (k := l) (n := 0) le_rfl le_rfl f x
+      apply (integral_mono (by simpa using f.integrable_pow_mul (μ := μ) 0) _ h).trans
+      · rw [integral_mul_right, ← mul_assoc]
+        gcongr ?_ * ?_
+        · rw [mul_comm]
+        · apply rfl.le
```
Comment (loefflerd) [inline Mathlib/Analysis/Distribution/SchwartzSpace.lean:1079]: What are the advantages of `apply rfl.le` over `rfl` here?

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean [hunk 4]
```diff
@@ -1003,17 +1017,85 @@ theorem iteratedPDeriv_eq_iteratedFDeriv {n : ℕ} {m : Fin n → E} {f : 𝓢(E
...
+/-- The integral as a continuous linear map from Schwartz space to the codomain. -/
+def integral (μ : Measure D) [IsAddHaarMeasure μ] : 𝓢(D, V) →L[ℝ] V :=
+  mkCLMtoNormedSpace (fun f ↦ ∫ x, f x ∂μ)
+    (fun f g ↦ integral_add f.integrable g.integrable)
+    (fun a f ↦ integral_smul a f)
+    (by
+      let l := finrank ℝ D + 1
+      let m := (l, 0)
+      use Finset.Iic (l, 0)
+      use 2^l * ∫ x : D, (1 + ‖x‖)^(- (l : ℝ)) ∂μ
```
Comment (loefflerd) [inline Mathlib/Analysis/Distribution/SchwartzSpace.lean:1064]:
```suggestion
      exists Finset.Iic m, 2 ^ l * ∫ x : D, (1 + ‖x‖) ^ (-(l : ℝ)) ∂μ
```

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean [hunk 5]
```diff
@@ -1036,40 +1118,21 @@ variable (𝕜 E F)
...
-      have : MulAction NNReal (Seminorm 𝕜 𝓢(E, F)) := Seminorm.instDistribMulAction.toMulAction
-      simp only [Seminorm.comp_apply, coe_normSeminorm, Finset.sup_singleton,
-        schwartzSeminormFamily_apply_zero, Seminorm.smul_apply, one_smul, ge_iff_le,
-        BoundedContinuousFunction.norm_le (apply_nonneg _ _)]
-      exact norm_le_seminorm 𝕜 _ }
+  mkCLMtoNormedSpace toBoundedContinuousFunction (by intro f g; ext; exact add_apply)
+    (by intro a f; ext; exact smul_apply)
+    (⟨{0}, 1, zero_le_one, by
+      simpa [BoundedContinuousFunction.norm_le (apply_nonneg _ _)] using norm_le_seminorm 𝕜 ⟩)
+#align schwartz_map.to_bounded_continuous_function_lm SchwartzMap.toBoundedContinuousFunctionCLM
```
Comment (loefflerd) [inline Mathlib/Analysis/Distribution/SchwartzSpace.lean:1128]: Does this work (to align two different mathlib3 declarations to a single mathlib4 declaration)? I am surprised.

**Review by loefflerd — COMMENTED** [review]
This looks very nice, I'm sure the ` mkCLMtoNormedSpace` will have lots of other uses beyond this. I've left some minor formatting corrections. Otherwise LGTM.

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean (hunk 1 above)
Comment (mcdoll) [reply Mathlib/Analysis/Distribution/SchwartzSpace.lean:1056]: I would rather remove some of the 'CLM's in names where it is no longer necessary.

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean (hunk 5 above)
Comment (mcdoll) [reply Mathlib/Analysis/Distribution/SchwartzSpace.lean:1128]: it seems to work

File: Mathlib/Analysis/Distribution/SchwartzSpace.lean (hunk 1 above)
Comment (loefflerd) [reply Mathlib/Analysis/Distribution/SchwartzSpace.lean:1056]: What does "necessary" mean, though? It's a pretty widespread convention in mathlib that when you're wrapping up some relatively simple operation into a more complicated homomorphism class, this is indicated with a suffix to the name indicating what you're packaging it as; e.g. `Nat.castRingHom`, or `ContinuousMap.evalCLM`. I'd definitely be happier if this were adhered to here. (We can raise this one on zulip if you like.)

Comment (mcdoll) [reply Mathlib/Analysis/Distribution/SchwartzSpace.lean:1056]: By necessary I mean that there is a conflicting definition for the bare function. I don't have too strong opinions on that and if you prefer `integralCLM` I can rename it.

Comment (loefflerd) [reply Mathlib/Analysis/Distribution/SchwartzSpace.lean:1056]: But there is a conflict, isn't there? If you have `SchwartzMap` and `MeasureTheory` open at the same time (not a totally unlikely prospect) then you get a conflict between `SchwartzMap.integral` and `MeasureTheory.integral`.

**Comment by loefflerd** [comment]
Looks great!

maintainer merge

**Pull request closed without merging**